From ce27e910d3a80a3addd8c92a700babb9e5764dac Mon Sep 17 00:00:00 2001
From: Anoob Joseph <anoobj@marvell.com>
Date: Mon, 13 May 2024 06:38:06 +0000
Subject: [PATCH 338/513] crypto/cnxk: remove init of CPT result field in
 packet

The packet would be posted to CPT only when there is a valid result.
Skip setting of the same.

Signed-off-by: Anoob Joseph <anoobj@marvell.com>
Change-Id: I7ba981d944424de5e97defa45a35d32a6c98be9d
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/127718
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Base-Tests: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Reviewed-by: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/128315
---
 drivers/crypto/cnxk/cn10k_cryptodev_ops.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/crypto/cnxk/cn10k_cryptodev_ops.c b/drivers/crypto/cnxk/cn10k_cryptodev_ops.c
index d36516735a0b8..1108a8a1da4fc 100644
--- a/drivers/crypto/cnxk/cn10k_cryptodev_ops.c
+++ b/drivers/crypto/cnxk/cn10k_cryptodev_ops.c
@@ -1410,10 +1410,6 @@ cn10k_cryptodev_sec_inb_rx_inject(void *dev, struct rte_mbuf **pkts,
 	uint64_t dptr;
 	int i;
 
-	const union cpt_res_s res = {
-		.cn10k.compcode = CPT_COMP_NOT_DONE,
-	};
-
 	vf = cdev->data->dev_private;
 
 	lmt_base = vf->rx_inj_lmtline.lmt_base;
@@ -1488,9 +1484,6 @@ cn10k_cryptodev_sec_inb_rx_inject(void *dev, struct rte_mbuf **pkts,
 		inst_67 = vsetq_lane_u64(u64_1, inst_67, 1);
 		vst1q_u64(&inst->w6.u64, inst_67);
 
-		rte_atomic_store_explicit((unsigned long __rte_atomic *)&hw_res->u64[0], res.u64[0],
-					  rte_memory_order_relaxed);
-
 		inst += 2;
 	}
 
-- 
2.25.1

